### DISCUSSION TITLE
Download Zip used to be a link but isn't any more, so I can't copy the URL

### DISCUSSION THREAD

**Discussion by simonw** [community/community#84699] [Repositories] [answered]
### Select Topic Area

Bug

### Body

I frequently need to grab the URL to the zip file for a repo or branch, so that I can paste it into `pip install` or tooling like that.

I used to be able to do that by navigating to the "Download ZIP" link, right clicking on it and clicking "Copy Link" (in Firefox).

That doesn't work any more!

![CleanShot 2024-01-05 at 15 23 20@2x](https://github.com/community/community/assets/9599/aeb84219-dfe3-4e8a-82e9-4a5667179953)

From the browser DevTools it looks like there's now some kind of React-based "event" thing going on:

![CleanShot 2024-01-05 at 15 26 18@2x](https://github.com/community/community/assets/9599/6113e53f-b719-48a2-bbd6-4888d4a681e4)

It would be great if this could go back to being a `<a href="...">` so I can get that URL. Or you could make the URL available with a copy button, that would work too.

**Comment by darkotodoric** [comment]
Not sure why GitHub changed from a link to a button, probably they want people to visit GitHub. In that way, they collect more data from that visitor compared to the situation when someone just downloads ZIP where they don't have that match data to collect.

You can share the link now, but you will have to "make" it yourself

```
https://github.com/{username}/{repository}/archive/main.zip
```
example:
```
https://github.com/darkotodoric/efficient-file-management/archive/main.zip
```

**Comment by samvv** [comment]
This is really annoying if you're using the terminal very frequently. Shame GitHub decided to change this.

**Comment by simonw** [comment]
I built myself an Observable notebook interface to work around this problem: https://observablehq.com/@simonw/download-github-repo

![Animated demo. Notebook is called URL to downloaed a GitHub repository as a zip file. You enter the URL to the repo (or in my case I typed simonw/datasette) and it gives you a URL to a zip, which starts with codeload.github.com. Clicking the "Use the latest commit hash" button fetches the latest commit hash and shows you that URL instead.](https://github.com/community/community/assets/9599/930b5a8d-e8c0-4f00-9550-8b40cfc30e51)

**Comment by simonw** [comment, marked as answer]
Fantastic news: this bug has just been fixed!
![Screenshot of a GitHub repository page with the Code green button pressed to show the Local panel, which now has a "Copy Link" option if you right click on the Download ZIP button](https://github.com/community/community/assets/9599/37a8b6bc-6fee-41bd-943e-2aa3de632f93)

**Comment by moajez** [comment]
thanks man, it works
